packages/harmony/src/components/select/Select/types.ts

```diff
@@ -1,68 +1,44 @@
-import { IconComponent } from 'components/icon'
-import { Origin } from 'components/popup/types'
+import { ReactNode } from 'react'
 
-import { SelectInputProps } from '../SelectInput/types'
+import { IconComponent } from 'components/icon'
+import { TextInputProps } from 'components/input'
+import { MenuProps } from 'components/internal/Menu'
 
-export type SelectOption = {
-  value: string
+export type SelectOption<Value extends string> = {
+  value: Value
   /**
    * The label to display. If not provided, uses the value.
    */
   label?: string
-  helperText?: string
   icon?: IconComponent
   leadingElement?: JSX.Element
 }
 
-export type SelectPopupProps = {
-  /**
-   * Popup anchor origin
-   * @default { horizontal: 'left', vertical: 'bottom' }
-   */
-  popupAnchorOrigin?: Origin
-
-  /**
-   * Popup max height
-   */
-  popupMaxHeight?: number
-
-  /**
-   * Popup transform origin
-   * @default { horizontal: 'left', vertical: 'top' }
-   */
-  popupTransformOrigin?: Origin
-
+type ChildrenProps<Value> = {
   /**
-   * Popup portal location passed to the inner popup
+   * A function to handle when the value is changed
    */
-  popupPortalLocation?: HTMLElement
-
-  /**
-   * zIndex applied to the inner Popup component
-   */
-  popupZIndex?: number
+  onChange: (value: Value) => void
+  options: ReactNode
```

Comment:
idk if either is necessary, but odd to have one docstring and not the other, and below in SelectProps, it's the opposite